File: packages/session-recorder-react-native/ios/SessionRecorderNativeSpec.swift
import Foundation
import React

@objc(SessionRecorderNativeSpec)
class SessionRecorderNativeSpec: NSObject, RCTTurboModule {

  @objc static func moduleName() -> String! {
    return "SessionRecorderNative"
  }

  @objc static func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc func captureAndMask(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
    // Implementation will be provided by the actual module
    reject("NOT_IMPLEMENTED", "Method not implemented", nil)
  }

  @objc func captureAndMaskWithOptions(_ options: NSDictionary, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
    // Implementation will be provided by the actual module
    reject("NOT_IMPLEMENTED", "Method not implemented", nil)
  }

  @objc func startGestureRecording(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
    // Implementation will be provided by the actual module
    reject("NOT_IMPLEMENTED", "Method not implemented", nil)
  }

  @objc func stopGestureRecording(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
    // Implementation will be provided by the actual module
    reject("NOT_IMPLEMENTED", "Method not implemented", nil)
  }

  @objc func isGestureRecordingActive(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
    // Implementation will be provided by the actual module
    reject("NOT_IMPLEMENTED", "Method not implemented", nil)
  }

  @objc func setGestureCallback(_ callback: @escaping RCTResponseSenderBlock) {
    // Implementation will be provided by the actual module
  }

  @objc func recordGesture(_ gestureType: String, x: NSNumber, y: NSNumber, target: String?, metadata: NSDictionary?) {
    // Implementation will be provided by the actual module
  }

  @objc func addListener(_ eventName: String) {
    // Required for RN event emitter contracts
  }

  @objc func removeListeners(_ count: Int) {
    // Required for RN event emitter contracts
  }
}

File: packages/session-recorder-react-native/android/src/main/java/com/sessionrecordernative/SessionRecorderNativeSpec.kt
package com.multiplayer.sessionrecordernative

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.Callback
import com.facebook.react.bridge.ReadableMap
import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.WritableMap

/**
 * Spec class for SessionRecorderNative TurboModule
 * This class defines the interface that React Native expects for TurboModules
 */
class SessionRecorderNativeSpec(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String {
        return NAME
    }

    companion object {
        const val NAME = "SessionRecorderNative"
    }

    // Delegate to the actual implementation
    private val implementation = SessionRecorderNativeModule(reactContext)

    @ReactMethod
    fun captureAndMask(promise: Promise) {
        implementation.captureAndMask(promise)
    }

    @ReactMethod
    fun captureAndMaskWithOptions(options: ReadableMap, promise: Promise) {
        implementation.captureAndMaskWithOptions(options, promise)
    }

    @ReactMethod
    fun startGestureRecording(promise: Promise) {
        implementation.startGestureRecording(promise)
    }

    @ReactMethod
    fun stopGestureRecording(promise: Promise) {
        implementation.stopGestureRecording(promise)
    }

    @ReactMethod
    fun isGestureRecordingActive(promise: Promise) {
        implementation.isGestureRecordingActive(promise)
    }

    @ReactMethod
    fun setGestureCallback(callback: Callback) {
        implementation.setGestureCallback(callback)
    }

    @ReactMethod
    fun recordGesture(
        gestureType: String,
        x: Double,
        y: Double,
        target: String?,
        metadata: ReadableMap?
    ) {
        implementation.recordGesture(gestureType, x, y, target, metadata)
    }

    @ReactMethod
    fun addListener(eventName: String) {
        implementation.addListener(eventName)
    }

    @ReactMethod
    fun removeListeners(count: Int) {
        implementation.removeListeners(count)
    }
}
